src/wp-includes/media.php

```diff
@@ -1985,6 +1975,18 @@ function wp_filter_content_tags( $content, $context = null ) {
  * @return string The filtered image tag markup.
  */
 function wp_img_tag_add_auto_sizes( string $image ): string {
+	/**
+	 * Filters whether auto-sizes for lazy loaded images is enabled.
+	 *
+	 * @since 6.7.1
+	 *
+	 * @param boolean $enabled Whether auto-sizes for lazy loaded images is enabled.
+	 * @param string  $image   The image tag markup being modified.
```

Comment:
Passing image markup as context allows this to be filtered based on the specific image, which is much more precise.

Reply:
We may want to align this with the way the lazy-loading filters work. There is one general "feature enabled" filter, and another one for "whether to add it for _this_ image".

The filter here would be more about whether to add it for the particular image. To be fair, that's enough on its own, if we don't want to introduce a separate generic "is the sizes=auto feature enabled?" filter. But I'd prefer that we then name the filter consistently with the equivalent lazy-loading filter, e.g. `wp_img_tag_add_auto_sizes`. That would also align with the function name.

Reply:
Renamed for consistency in bbaed9c.

Reply:
See above comment regarding whether or not we should pass image markup (for now). In the future we definitely should, though I'm not sure how crucial that is for a 6.7.1 hot fix filter.